### PR TITLE
Ocsp client

### PR DESCRIPTION
**Pull request by felipemuhamed**

### CODE REVIEW

File: examples/ocsp_example.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+           (example: http://172.18.0.2:8080/ejbca/publicweb/status/ocsp)...';
+let cn = 'insert here the CN of your client...';
+let path = 'insert here the path to be saved the client.crt...';
+
+const ocsp = new OCSPRequester.OCSPRequester();
+
+ocsp.sendRequest(url, mycert, cacert, cn, path).then((isAuth) => {
+    if (isAuth) {
+        /* The client is authorized */
+        console.log('Authorized');
```
Comment (mprevide) [inline examples/ocsp_example.js]: Please, use var logger = require("@dojot/dojot-module-logger").logger with a TAG

File: examples/ocsp_example.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    else {
+        /* The client is not authorized */
+
+        console.log('Not authorized');
```
Comment (mprevide) [inline examples/ocsp_example.js]: Logger

File: lib/ocsp.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    _createClientCRT(clientCert, path) {
+
+        /* format the client cert to be used */
+        let cert = "-----BEGIN CERTIFICATE-----" + "\n"
+            + clientCert +
+            "\n" + "-----END CERTIFICATE-----"
+
+        /* save in a path*/
+        return new Promise((resolve, reject) => {
```
Comment (mprevide) [inline lib/ocsp.js]: I think it is not necessary to write to disk every time the certificate, you can use Buffer.from and create a buffer with the header.

File: lib/ocsp.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    sendRequest(url, mycert, cacert, cn, clientPath) {
+        return new Promise((resolve, reject) => {
+
+            this._createClientCRT(mycert, clientPath).then(() => {
+                openssl(['ocsp', '-issuer', cacert, '-cert',
+                    clientPath, '-url', url],
+                    (err, buffer) => {
+
+                        let bufferString = buffer.toString();
+                        console.log(bufferString);
```
Comment (mprevide) [inline lib/ocsp.js]: Remove this console.log

File: tests/unit-tests/ocsp.test.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    describe("When an OCSP request was sent", () => {
+        it('returns false if cert is not good', () => {
+            jest.mock('openssl-nodejs', () =>
+                (args, callback) =>
+                    callback("error", "../certs/5ef927.crt: revoke\nThis Update: Jul 27 23:28:19 2019 GMT"));
+
+            const OCSPRequester = require('../../lib/ocsp.js');
+
+            const ocsp = new OCSPRequester.OCSPRequester();
+            ocsp._createClientCRT = jest.fn(() => { return Promise.resolve('test') });
```
Comment (mprevide) [inline tests/unit-tests/ocsp.test.js]: Are you testing within the callback?

Comment (muhamedcpqd) [reply tests/unit-tests/ocsp.test.js]: In this part, i'm testing the sendRequest() method. However this method is inside _createClientCRT() that has other dependencies like fs that i want to mock. To test only sendRequest(), i decided to mock _createClientCRT() entirely..

File: examples/ocsp_example.js (hunk 1 above)
Comment (mprevide) [reply examples/ocsp_example.js]: You can add the logger here too, as an example of good practice. :)

**Comment by mprevide** [comment]
Is not necessery add openssl install in dockerfile?

**Comment by muhamedcpqd** [comment]
@mprevide this lib doesn't implements an image, just a node module. So the openssl instalation package is in charge of the final user.

**Comment by mprevide** [comment]
It's true xD
Sorry =)

**Comment by mprevide** [comment]
Perhaps describe in the Readme about the need for OpenSSL and explain how to use ocsp.

**Comment by mprevide** [comment]
Also see the need to publish to npm at this time.

**Comment by muhamedcpqd** [comment]
> Perhaps describe in the Readme about the need for OpenSSL and explain how to use ocsp.

Good idea @mprevide ! i will do it.

**Comment by muhamedcpqd** [comment]
> Also see the need to publish to npm at this time.

I think we can wait a little more to close all the pull requests and then create a new npm version. I dont remember correctly who is the owner of this module (maybe its me lol). @giovannicuriel , the owner its you? =]

**Comment by giovannicuriel** [comment]
Probably yes.

**Comment by mprevide** [comment]
Maybe can you close this PR?

**Pull request closed without merging**